.github/actions/install_requirements/action.yml

```diff
@@ -15,7 +15,7 @@ runs:
       run: |
         PYTHON_VERSION="${{ inputs.python-version }}"
         if [ $PYTHON_VERSION == "dev" ]; then
-          PYTHON_VERSION=$(sed -n "s/ARG PYTHON_VERSION=//p" Dockerfile)
+          PYTHON_VERSION=$(sed -Ene "s/ARG PYTHON_VERSION=([0-9\.]+).*/\1/p" Dockerfile)
```

Comment:
I really should have put a comment here to say what it is doing, but it definitely needs one now there's a regex in it!

Reply:
I'd like to get rid of this action altogether if at all possible, so that we're not maintaining an action alongside the copier template. Is there reason that the old matrix style from the skeleton wasn't kept up?

```yaml
  test:
    if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.repository
    strategy:
      fail-fast: false
      matrix:
        os: ["ubuntu-latest"] # can add windows-latest, macos-latest
        python: ["3.9", "3.10", "3.11"]
        install: ["-e .[dev]"]
        # Make one version be non-editable to test both paths of version code
        include:
          - os: "ubuntu-latest"
            python: "3.8"
            install: ".[dev]"
```